### PR TITLE
taha/LP-1607 Implement the verify certificate page

### PR DESCRIPTION
**Pull request by arbisofttaha**

### CODE REVIEW

File: openedx/features/student_certificates/views.py [hunk 1]
```diff
@@ -195,6 +195,34 @@ def download_certificate_pdf(request, certificate_uuid):
...
+        The verify certificate response.
+
+    """
+
     certificate_verification_key_obj = get_object_or_404(CertificateVerificationKey, verification_key=key)
     certificate_uuid = certificate_verification_key_obj.generated_certificate.verify_uuid
-    return redirect(reverse('shared_achievements', kwargs={'certificate_uuid': certificate_uuid}))
+
+    try:
+        certificate = GeneratedCertificate.eligible_certificates.get(
```
Comment (emraan004) [inline openedx/features/student_certificates/views.py]: We already have generated certificate in above statement at line 211, can we use that instead re-querying for it?

File: openedx/features/student_certificates/views.py [hunk 2]
```diff
@@ -195,6 +195,34 @@ def download_certificate_pdf(request, certificate_uuid):
...
+            verify_uuid=certificate_uuid,
+            status=CertificateStatuses.downloadable
+        )
+    except GeneratedCertificate.DoesNotExist:
+        raise Http404
+
+    context = {
+        'achieved_by': certificate.user.first_name + ' ' + certificate.user.last_name,
```
Comment (emraan004) [inline openedx/features/student_certificates/views.py]: let's use format function to concatenate first name and last name or see if we have a function that returns full name

File: openedx/features/student_certificates/views.py [hunk 3]
```diff
@@ -195,6 +195,26 @@ def download_certificate_pdf(request, certificate_uuid):
...
+    """
+    Provides verify certificate page
+
+    Arguments:
+        request: The request object.
+        key: Verification key of certificate.
+
+    Returns:
+        The verify certificate response.
+
```
Comment (zeshanarifios) [inline openedx/features/student_certificates/views.py]: please remove this extra space.

File: openedx/features/student_certificates/views.py [hunk 4]
```diff
@@ -195,6 +195,26 @@ def download_certificate_pdf(request, certificate_uuid):
 
 
 def verify_certificate(request, key):
+    """
+    Provides verify certificate page
+
+    Arguments:
+        request: The request object.
+        key: Verification key of certificate.
+
```
Comment (zeshanarifios) [inline openedx/features/student_certificates/views.py]: remove extra space

File: openedx/features/student_certificates/views.py [hunk 5]
```diff
@@ -195,6 +195,26 @@ def download_certificate_pdf(request, certificate_uuid):
...
+    """
+
     certificate_verification_key_obj = get_object_or_404(CertificateVerificationKey, verification_key=key)
-    certificate_uuid = certificate_verification_key_obj.generated_certificate.verify_uuid
-    return redirect(reverse('shared_achievements', kwargs={'certificate_uuid': certificate_uuid}))
+    certificate = certificate_verification_key_obj.generated_certificate
+
+    context = {
+        'achieved_by': certificate.user.get_full_name(),
+        'achieved_at': certificate.created_date.strftime("%B %d, %Y"),
```
Comment (zeshanarifios) [inline openedx/features/student_certificates/views.py]: use single quotes to make quotes consistent.

File: openedx/features/student_certificates/views.py (hunk 3 above)
Comment (arbisofttaha) [reply openedx/features/student_certificates/views.py]: Done

File: openedx/features/student_certificates/views.py (hunk 4 above)
Comment (arbisofttaha) [reply openedx/features/student_certificates/views.py]: Done

File: openedx/features/student_certificates/views.py (hunk 5 above)
Comment (arbisofttaha) [reply openedx/features/student_certificates/views.py]: Done

**Pull request merged**